### DISCUSSION TITLE
[Python] Django+uwsgi Agent  access failure

### DISCUSSION THREAD

**Discussion by WinterWaterWarm** [apache/skywalking#10348] [Q&A] [answered]
Agent started at `wsgi.py`
![企业微信截图_16756531694096](https://user-images.githubusercontent.com/39685919/216876451-e73d5c07-3329-4737-a860-e3eee1d6040b.png)

Then error raised at uwsgi's log
![企业微信截图_16756536278276](https://user-images.githubusercontent.com/39685919/216876606-a10c01d7-eee6-4635-b6b4-d3137c693584.png)

what might be the reason for this error raising?

**Comment by Superskyyy** [comment]
@WinterWaterWarm Hello, this doesn't seem to be from our side (it originates from google's protobuf, just chain of imports). And I haven't seen anyone encounter this error yet, I guess can it be due to the import order? Please try to move the import to top and see if anything changes.

Please also provide your environment information, is there something special?

**Comment by Superskyyy** [comment]
@WinterWaterWarm Check this one out and see if it helps. https://groups.google.com/g/protobuf/c/YRW_TROq2Rg/m/dlW5QoE9DAAJ

**Comment by Superskyyy** [comment, marked as answer]
@WinterWaterWarm Ha, can you try move the import inside of @postfork instead of having them outside? Maybe it will resolve the problem.

From [here ](https://www.pythonanywhere.com/forums/topic/31463/)I can infer the root cause is related to Google's protobuf package cannot be imported twice in the same process, then moving the import into postfork sounds like a fix.

**Reply by WinterWaterWarm** [reply]
Thanks for your help!
It worked.The new code as follows
![企业微信截图_16757544779342](https://user-images.githubusercontent.com/39685919/217175353-0ae04f3f-fb18-4f3a-9249-536ff5a65d4f.png)
As for why I have this situation, it may be related to my container configuration. 
I have to take time to find it out.
